controllers/lvmcluster_controller.go

```diff
@@ -220,6 +227,71 @@ func (r *LVMClusterReconciler) reconcile(ctx context.Context, instance *lvmv1alp
 	return ctrl.Result{}, nil
 }
 
+func (r *LVMClusterReconciler) verifyLvmClusterSpec(ctx context.Context, instance *lvmv1alpha1.LVMCluster) error {
+
+	err := r.verifyLvmClusterDeviceClasses(ctx, instance)
+	if err != nil {
+		// Apply status changes
+		updateErr := r.Client.Status().Update(ctx, instance)
+		if updateErr != nil {
+			r.Log.Error(updateErr, "failed to update LVMCluster status")
+			return updateErr
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (r *LVMClusterReconciler) verifyLvmClusterDeviceClasses(ctx context.Context, instance *lvmv1alpha1.LVMCluster) error {
+
+	// make sure no device overlap with another VGs
+	// use map to find the duplicate entries for paths
+	/*
+		{
+		  "nodeSelector1": {
+		        "/dev/sda": "vg1",
+		        "/dev/sdb": "vg1"
+		    },
+		    "nodeSelector2": {
+		        "/dev/sda": "vg1",
+		        "/dev/sdb": "vg1"
+		    }
+		}
+	*/
+	devices := make(map[string]map[string]string)
+
+	for _, deviceClass := range instance.Spec.Storage.DeviceClasses {
+		if deviceClass.DeviceSelector != nil {
```

Comment:
what if we (in the future) have two deviceClasses - one taking up all the disks and one with manual disk selection?

Reply:
we should not support such cases IMO. This code will just check the manual disk paths only as other device class wont have any list of devices in the spec.